[quant][graphmode] API - add more passes to graph mode

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27119 [WIP][quant][graphmode] Prepack folding for conv2d
* #27118 [quant][graphmode] Add insert_prepack_unpack for conv2d
* #27103 [quant][graphmode] Enable tests
* #27102 [quant][graphmode][fix] insert_prepack_unpack pattern
* **#27093 [quant][graphmode] API - add more passes to graph mode**

Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678262](https://our.internmc.facebook.com/intern/diff/D17678262)

jerryzh168 committed:
[quant][graphmode] API - add more passes to graph mode

Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] API - add more passes to graph mode"


Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] API - add more passes to graph mode"


Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good, but please in future follow the the next guidelines:
1) Make sure the change doesn't break tests
2) Make sure the change is tested
3) If the change relies on a feature that will be landed later, there is no value in landing this change before that feature. There is not much value in dummy tests that always pass or are disabled.

Currently this change isn't exercised by any test which is not great.

jerryzh168 commented:
yeah previous tests was skipped because there are some known issues to fix

jerryzh168 committed:
Update on "[quant][graphmode] API - add more passes to graph mode"


Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678262](https://our.internmc.facebook.com/intern/diff/D17678262)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] API - add more passes to graph mode"


Summary:
Add `insert_prepack_unpack` and `fold_prepack` to `convert_script`

Test Plan:
.

Reviewers:
pt1quant

Subscribers:

Tasks:

Tags:

Differential Revision: [D17678262](https://our.internmc.facebook.com/intern/diff/D17678262)

[ghstack-poisoned]
